### DISCUSSION TITLE
CocoaPods 都做了什么？ · /cocoapods

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#273] [Announcements]
https://draveness.me/cocoapods

**Comment by denglitong** [comment]
面向信仰编程，这个签名不错

**Comment by draveness** [comment]
> @denglitong
> 面向信仰编程，这个签名不错

我觉得 ok

**Comment by RiberWang** [comment]
贼6 
就是看的不太懂

**Comment by ghost** [comment]
有必要去learn一下Ruby，再来回看本文

**Comment by lanyasheng** [comment]
很强

**Comment by chenzhengxu** [comment]
你好，我刚开始阅读CocoaPods项目，想问下这边lib和spec的职责分别是什么，看lib部分的代码应该是主要实现代码，而spec部分的代码让我有些困惑

![img](https://note.youdao.com/yws/public/resource/e0e8fbe2fd8f25daf46654a229d011c2/xmlnote/WEBRESOURCE3d65f188cf79317a6389b527c6098ed3/1543)

**Comment by Desgard** [comment]
> @chenzhengxu
> 你好，我刚开始阅读CocoaPods项目，想问下这边lib和spec的职责分别是什么，看lib部分的代码应该是主要实现代码，而spec部分的代码让我有些困惑
> 
> ![img](https://note.youdao.com/yws/public/resource/e0e8fbe2fd8f25daf46654a229d011c2/xmlnote/WEBRESOURCE3d65f188cf79317a6389b527c6098ed3/1543)

一般的 Ruby 项目，lib 目录包括了所有的项目代码，也就是常说的 **扩展模块**，而 spec 中一般是 **测试模块**。看 ror 更为清晰。
